app/assets/javascripts/discourse/controllers/forgot-password.js.es6

```diff
@@ -52,6 +52,42 @@ export default Ember.Controller.extend(ModalFunctionality, {
       });
 
       return false;
+    },
+
+    emailLogin() {
+      if (this.get('submitDisabled')) return false;
+
+      this.set('disabled', true);
+
+      ajax('/u/email-login', {
+        data: { login: this.get('accountEmailOrUsername').trim() },
+        type: 'POST'
+      }).then(data => {
+        const escaped = escapeExpression(this.get('accountEmailOrUsername'));
+        const isEmail = this.get('accountEmailOrUsername').match(/@/);
+        let key = 'email_login.complete_' + (isEmail ? 'email' : 'username');
+        let extraClass;
+
+        if (data.user_found === true) {
+          key += '_found';
+          this.set('accountEmailOrUsername', '');
+          bootbox.alert(I18n.t(key, {email: escaped, username: escaped}));
+          this.send("closeModal");
+        } else {
+          if (data.user_found === false) {
+            key += '_not_found';
+            extraClass = 'error';
+          }
+
+          this.flash(I18n.t(key, {email: escaped, username: escaped}), extraClass);
+        }
+      }).catch(e => {
+        this.flash(extractError(e), 'error');
```

Comment:
It's a modal so the error is placed on the top of it. `popupAjaxError` takes the error from `extractError` and shows as a dialog.